### PR TITLE
IG-14164: Limit size of TSDB frames.

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: test/tsdb_integration_test.go [hunk 1]
```diff
@@ -313,7 +327,7 @@ func (tsdbSuite *TsdbTestSuite) TestDeleteWithTimestamp() {
 		Backend: tsdbSuite.backendName,
 		Table:   table,
 		Start:   "0",
-		End:     "now",
+		End:     "157728271500000", // Very high upper bound to catch all samples
```
Comment (talIguaz) [inline test/tsdb_integration_test.go]: use a variable

File: test/tsdb_integration_test.go [hunk 2]
```diff
@@ -32,26 +33,34 @@ func GetTsdbTestsConstructorFunc() SuiteCreateFunc {
 	}
 }
 
-func (tsdbSuite *TsdbTestSuite) generateSampleFrame(t testing.TB) frames.Frame {
+func (tsdbSuite *TsdbTestSuite) generateSampleFrame(t testing.TB, anchorTime time.Time) frames.Frame {
```
Comment (talIguaz) [inline test/tsdb_integration_test.go]: the same as `generateSampleFrameWithEndTime` can delete one of them.

File: test/tsdb_integration_test.go [hunk 3]
```diff
@@ -154,23 +164,26 @@ func (tsdbSuite *TsdbTestSuite) TestAll() {
...
 	for it.Next() {
-		// TODO: More checks
 		fr := it.At()
-		if !(fr.Len() == frame.Len() || fr.Len()-1 == frame.Len()) {
-			tsdbSuite.T().Fatalf("wrong length: %d != %d", fr.Len(), frame.Len())
-		}
+		resultCount += fr.Len()
+	}
+	// TODO: More checks
+	if !(resultCount == frame.Len() || resultCount-1 == frame.Len()) {
```
Comment (talIguaz) [inline test/tsdb_integration_test.go]: why are we checking `resultCount -1` ?

Comment (gtopper) [reply test/tsdb_integration_test.go]: No clue.

**Pull request merged**